src/lib/libpthread.js

```diff
@@ -435,11 +435,29 @@ var LibraryPThread = {
         worker = new Worker(p.createScriptURL('ignored'), {{{ pthreadWorkerOptions }}});
       } else
 #endif
-      // We need to generate the URL with import.meta.url as the base URL of the JS file
-      // instead of just using new URL(import.meta.url) because bundler's only recognize
-      // the first case in their bundling step. The latter ends up producing an invalid
-      // URL to import from the server (e.g., for webpack the file:// path).
-      worker = new Worker(new URL('{{{ TARGET_JS_NAME }}}', import.meta.url), {{{ pthreadWorkerOptions }}});
+      {
+#if expectToReceiveOnModule('mainScriptUrlOrBlob')
+        if (Module['mainScriptUrlOrBlob']) {
+          var pthreadMainJs = Module['mainScriptUrlOrBlob'];
+          if (typeof pthreadMainJs != 'string') {
+            pthreadMainJs = URL.createObjectURL(pthreadMainJs);
+          }
+          worker = new Worker(new URL('{{{ TARGET_JS_NAME }}}', import.meta.url), {{{ pthreadWorkerOptions }}});
+        } else
+#endif
+        {
+          if (import.meta.url.startsWith('blob:')) {
+            // If our import.meta.url is a blob, we should just use it.
+            worker = new Worker(import.meta.url, {{{ pthreadWorkerOptions }}});
+          } else {
+            // We need to generate the URL with import.meta.url as the base URL of the JS file
+            // instead of just using new URL(import.meta.url) because bundler's only recognize
+            // the first case in their bundling step. The latter ends up producing an invalid
+            // URL to import from the server (e.g., for webpack the file:// path).
+            worker = new Worker(new URL('{{{ TARGET_JS_NAME }}}', import.meta.url), {{{ pthreadWorkerOptions }}});
```

Comment:
Man.. this limitation of the bundlers really makes our generated code nasty.   I wonder if the bundlers can be fixed... its really a shame to have all this extra stuff just to make them happy.

Reply:
I wonder if we could just fix webpack to support `new Worker(import.meta.url)` which is what we actually want to write here

Reply:
I found a webpack bug from way back about this: https://github.com/webpack/webpack/issues/12638

Reply:
My understanding from https://github.com/webpack/webpack/issues/12638#issuecomment-2698973729 is that this is fixed now, so what’s the move?  To keep the mainScriptUrlOrBlob path and default to new Worker(import.meta.url, options), eliminating the new URL calls/paths?

Reply:
Fantastic.   I think we should probably check with all the different bundlers, but assuming we can go with `new Worker(import.meta.ur)` that sounds like the best solution.

Re-adding support for mainScriptUrlOrBlob sounds good too.